### DISCUSSION TITLE
Custom Struct Tags

### DISCUSSION THREAD

**Discussion by ericsgagnon** [jackc/pgx#2167] [General]
Is it possible to configure the struct tag for functions like StructToRowByName? I'm moving data between systems, so db isn't informative enough and has conflicts with other systems and packages. ie - I have structs that are used for multiple end points and multiple sources (like mssql, oracle, postgres) so db tag becomes an issue.

**Comment by jackc** [comment]
It's not externally configurable. Internally, it would be trivial to change.

```go
const structTagKey = "db"
```

I suppose it could be made a public variable, but I'm generally not in favor of global configuration variables.

However, all the functions like `RowToStructByName` are simply implementations of `RowToFunc`. They don't have rely on any access to internal pgx functionality. You could copy the entire `RowToStructByName` type and change what you needed.
